### PR TITLE
Scenery Set Colour Game Actions

### PR DESCRIPTION
**Pull request by duncanspumpkin**
I'll put all of the set colour actions in this one PR.

### CODE REVIEW

File: src/openrct2/actions/BannerSetStyleAction.hpp [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+                else
+                {
+                    return MakeResult(GA_ERROR::INVALID_PARAMETERS, STR_ERR_CANT_SET_BANNER_TEXT);
+                }
+                break;
+            }
+            case BannerSetStyleType::NoEntry:
+            {
+                BannerElement* bannerElement = tileElement->AsBanner();
+                banner->flags &= BANNER_FLAG_NO_ENTRY;
```
Comment (ZehMatt) [inline src/openrct2/actions/BannerSetStyleAction.hpp]: Is this supposed to remove the flag?

File: src/openrct2/actions/BannerSetStyleAction.hpp [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+                banner->flags |= BANNER_FLAG_NO_ENTRY & (_parameter != 0);
```
Comment (ZehMatt) [inline src/openrct2/actions/BannerSetStyleAction.hpp]:
```suggestion
                banner->flags |= (parameter != 0) ? BANNER_FLAG_NO_ENTRY : 0
```
Would be slightly easier to read.

File: src/openrct2/actions/LargeScenerySetColourAction.hpp [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            {
+                if (!map_is_location_owned(currentTile.x, currentTile.y, currentTile.z))
+                {
+                    return MakeResult(GA_ERROR::NOT_OWNED, STR_CANT_REPAINT_THIS, STR_LAND_NOT_OWNED_BY_PARK);
+                }
+            }
+
+            if (isExecuting)
+            {
+                auto tileElement = map_get_large_scenery_segment(currentTile.x, currentTile.y, _loc.z / 8, _loc.direction, i);
```
Comment (ZehMatt) [inline src/openrct2/actions/LargeScenerySetColourAction.hpp]: Potentially null

File: src/openrct2/actions/BannerSetStyleAction.hpp (hunk 1 above)
Comment (duncanspumpkin) [reply src/openrct2/actions/BannerSetStyleAction.hpp]: Yes. Should be `~BANNER_FLAG_NO_ENTRY`

**Comment by ZehMatt** [comment]
Code wise looking okay, have to yet test them.

**Comment by duncanspumpkin** [comment]
Works fine on multiplayer. Tested all actions.

**Pull request merged**